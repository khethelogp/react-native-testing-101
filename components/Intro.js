import React, {Component, useState} from 'react';
import {Button, StyleSheet, Text, View} from 'react-native';

const Intro = () => {
  const [status, setStatus] = useState('');

  return (
    <View style={styles.container}>
      <Text style={styles.welcome}>Welcome to React Native!</Text>
      <Text style={styles.instructions}>
        This is a React Native snapshot test.
      </Text>

      <View
        style={{
          width: '100%',
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        }}>
        <Text testID="myText">{status}</Text>
        <Button
          testID="myButton"
          title="Press Me"
          onPress={() => {
            setStatus('Hi Mom!');
          }}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    flex: 1,
    justifyContent: 'center',
  },
  instructions: {
    color: '#333333',
    marginBottom: 5,
    textAlign: 'center',
  },
  welcome: {
    fontSize: 20,
    margin: 10,
    textAlign: 'center',
  },
});

export default Intro;
