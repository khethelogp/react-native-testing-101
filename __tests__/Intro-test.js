import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {ActivityIndicator, FlatList, Text, TextInput} from 'react-native';
import {Intro} from '../components';

const tree = renderer.create(<Intro />);

// unit test
test('renders correctly', () => {
  tree.toJSON();
  expect(tree).toMatchSnapshot();
});

/// unit state test
test('button press', () => {
  //press button
  const button = tree.root.findByProps({testID: 'myButton'}).props;
  act(() => button.onPress());

  //expect text to equal 'Hi Mom!'
  const text = tree.root.findByProps({testID: 'myText'}).props;
  expect(text.children).toEqual('Hi Mom!');
});

// Intergration test

it('renders the ActivityIndicator component', () => {
  const tree = renderer
    .create(<ActivityIndicator animating={true} size="small" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders the TextInput component', () => {
  const tree = renderer
    .create(<TextInput autoCorrect={false} value="apple banana kiwi" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the FlatList component', () => {
  const tree = renderer
    .create(
      <FlatList
        data={['apple', 'banana', 'kiwi']}
        keyExtractor={item => item}
        renderItem={({item}) => <Text>{item}</Text>}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
