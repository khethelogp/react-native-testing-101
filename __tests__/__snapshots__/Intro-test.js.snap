// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      testID="myText"
    />
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="myButton"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Press Me
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders the ActivityIndicator component 1`] = `
<ActivityIndicator
  animating={true}
  size="small"
/>
`;

exports[`renders the FlatList component 1`] = `
<RCTScrollView
  data={
    Array [
      "apple",
      "banana",
      "kiwi",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        apple
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        banana
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text>
        kiwi
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders the TextInput component 1`] = `
<TextInput
  autoCorrect={false}
  value="apple banana kiwi"
/>
`;
